packages/shell-api/src/shell-instance-state.ts

```diff
@@ -135,6 +140,14 @@ export interface ShellPlugin {
 // eslint-disable-next-line no-control-regex
 const CONTROL_CHAR_REGEXP = /[\x00-\x1F\x7F-\x9F]/g;
 
+function connectionIdFromURI(uri: string): string {
+  // turn the uri into something we can safely use as part of a "filename"
+  // inside autocomplete
+  const hash = crypto.createHash('sha256');
+  hash.update(uri);
+  return hash.digest('hex');
+}
```

Comment:
This feels like it miiiight be worth caching on the `Mongo` instance, not sure if that would be premature optimization but while hashing is cheap, it might not be do-a-few-times-per-autocomplete cheap?